File: ruby-python/python/examples/html.rb
require 'python'
require 'python/httplib'
require 'python/htmllib'
require 'python/formatter'

url = ARGV.shift
if url != nil && url =~ /^http:\/\/([^\/]+)(\/.*)$/
  host, path = $1, $2
else
  print "Usage: ruby ", $0, " http://host/[path]\n"
  exit(1)
end

h = Py::Httplib::HTTP.new(host)
h.putrequest('GET', path)
h.putheader('Accept', 'text/html')
h.putheader('Accept', 'text/plain')
h.endheaders()
errcode, errmsg, headers = h.getreply().to_a

if errcode == 200
  data = h.getfile().read()

  f = Py::Formatter::AbstractFormatter.new(Py::Formatter::DumbWriter.new())
  parser = Py::Htmllib::HTMLParser.new(f)
  parser.feed(data)
  parser.close()
else
  print errcode, ": Failed to fetch ", url, "\n"
  exit(1)
end

File: ruby-python/python/examples/html.py
import httplib
import htmllib
import formatter

import re, sys

match = None
if len(sys.argv) > 1:
    url = sys.argv[1]
    match = re.match('^http:\/\/([^\/]+)(\/.*)$', url)
if match:
    host, path = match.groups()
else:
    print "Usage: python ", sys.argv[0], " http://host/[path]"
    sys.exit(1)

h = httplib.HTTP(host)
h.putrequest('GET', path)
h.putheader('Accept', 'text/html')
h.putheader('Accept', 'text/plain')
h.endheaders()
errcode, errmsg, headers = h.getreply()

if errcode == 200:
    data = h.getfile().read()
    
    f = formatter.AbstractFormatter(formatter.DumbWriter())
    parser = htmllib.HTMLParser(f)
    parser.feed(data)
    parser.close()
else:
    print errcode, ": Failed to fetch", url
    sys.exit(1)
